modules/xfem/src/base/XFEMFuncs.C

```diff
@@ -814,4 +814,131 @@ crossProduct2D(const Point & point_a, const Point & point_b)
   return (point_a(0) * point_b(1) - point_b(0) * point_a(1));
 }
 
+Real
+pointSegmentDistance(const Point & x0, const Point & x1, const Point & x2, Point & xp)
+{
+  Point dx = x2 - x1;
+  Real m2 = dx * dx;
+  // find parameter value of closest point on segment
+  Real s12 = (x2 - x0) * dx / m2;
+  if (s12 < 0)
+    s12 = 0;
+  else if (s12 > 1)
+    s12 = 1;
+  // and find the distance
+  xp = s12 * x1 + (1 - s12) * x2;
+  return std::sqrt((x0 - xp) * (x0 - xp));
+}
+
+Real
+pointTriangleDistance(const Point & x0,
```

Comment:
Same here -- it's difficult for mortals like me to understand this algorithm. Either add some comments or add some unit tests, or both.

Reply:
I added a paper that I used for implementation. I hope it helps.